tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -429,88 +429,82 @@ pub fn logic_layer_aggregation(
                 }
 
                 let col_data = &col_data_map[&col_i];
-
-                let mut final_col_data: Vec<String> = vec![];
-                let mut i = 0;
-
-                for entry in col_data {
-                    if !exclude_row_indexes.contains(&i) {
-                        final_col_data.push(entry.clone());
-                    }
-
-                    i += 1;
-                }
+                let col_data: Vec<String> = col_data.iter()
+                    .enumerate()
+                    .filter(|&(i, _)| !exclude_row_indexes.contains(&i) )
+                    .map(|(_, e) | e.to_string())
+                    .collect();
 
                 // When returning data from multiple levels from the same
                 // hierarchy, there is a chance that this column will have
                 // multiple data types. In those cases, we will convert the
                 // whole column to string values.
                 if same_type {
-                    let mut column_data: ColumnData = ColumnData::Text(final_col_data.clone());
+                    let mut column_data: ColumnData = ColumnData::Text(col_data.clone());
```

Comment:
Would it make sense to change this block so that instead of having `let mut column_data: ColumnData = ColumnData::Text(col_data.clone());` we did something like `let column_data = match first_col.column_data {...` ? in the current case it looks like we might clone `col_data` and store that in column_data, even if we end up overwriting `column_data`?